src/Data/List/Relation/Unary/Any/Properties.agda

```diff
@@ -250,6 +250,19 @@ module _ {P : Pred A p} {Q : Pred B q} where
                                    y∈ys
              = lem₂ _ refl
 
+------------------------------------------------------------------------
+-- Half-applied product commutes with Any.
+
+module _ {R : REL A B r} where
+
+  Any-Σ⁺ʳ : ∀ {xs} → (∃ λ x → Any (flip R x) xs) → Any (∃ ∘ R) xs
+  Any-Σ⁺ʳ (b , here px) = here (b , px)
+  Any-Σ⁺ʳ (b , there pxs) = there (Any-Σ⁺ʳ (b , pxs))
+
+  Any-Σ⁻ʳ : ∀ {xs} → Any (∃ ∘ R) xs → ∃ λ x → Any (flip R x) xs
+  Any-Σ⁻ʳ (here (b , x)) = b , here x
+  Any-Σ⁻ʳ (there xs) = let b , xs' = Any-Σ⁻ʳ xs in b , there xs'
+
```

Comment:
Here you could make `R` infix instead and use sections rather than `flip R x`.
(Also worth adding extra parentheses in the pattern-matching let for clarity).

```agda
module _ {_~_ : REL A B r} where
￼	
￼	  Any-Σ⁺ʳ : ∀ {xs} → (∃ λ x → Any (_~ x) xs) → Any (∃ ∘ _~_) xs
￼	  Any-Σ⁺ʳ (b , here px)   = here (b , px)
￼	  Any-Σ⁺ʳ (b , there pxs) = there (Any-Σ⁺ʳ (b , pxs))
￼	
￼	  Any-Σ⁻ʳ : ∀ {xs} → Any (∃ ∘ _~_) xs → ∃ λ x → Any (_~ x) xs
￼	  Any-Σ⁻ʳ (here (b , x))  = b , here x
￼	  Any-Σ⁻ʳ (there xs)      = let (b , xs') = Any-Σ⁻ʳ xs in b , there xs'
```

Reply:
Done

Reply:
> Here you could make `R` infix instead and use sections rather than `flip R x`.

Done

> (Also worth adding extra parentheses in the pattern-matching let for clarity).

I'm not sure whether you meant about the pattern or the body, but see new version.

Reply:
Ah! Even better!